HW02_B_Mimic_with code.Rmd

```diff
@@ -0,0 +1,122 @@
+ggplot(diamonds, aes(x=cut, fill = clarity)) +
+    geom_bar(stat="count", position = "dodge") +
+    annotate("rect", xmin = 4.5, xmax= 5.5, ymin=0, ymax = 5000, alpha=0.2) + #this is to add a shaded rectangle
+    annotate("text", label = "My Best Diamonds,\nof course", x = 4, y = 4500)+ #this is to add the text 
+    labs(x="Diamond Cut", y = "Number of Diamonds", title = "My Diamond Collection", subtitle = "Boxplot representing the number of diamonds in my diamond collection \nby type of cut quality and clarity of diamond") + #adds labels to the plot
+    theme(plot.title = element_text(hjust = 0.5, face = "bold")) #this centers the title
+
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data(iris)
+```
+
+Using the iris dataset, make this graph:
+```{r graph 2 code, echo=FALSE}
+combined_plot <- ggplot(iris, aes(Sepal.Length, Petal.Length, color = Species, shape = Species)) +
+  geom_point() + #up to this step, R will produce a combined plot of all species
+  geom_smooth(method = "lm", se = FALSE, color = "black") #this will add linear fit with no errors
+combined_plot + facet_wrap(. ~ Species, scales = "free_y") #now we can separate the plot by species using facet_grid with formula x~y. I'm putting . for x and Species for y to have the dividing lines parallel to y-axis
+
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",]
+#install
+require("ggrepel") #useful for making text annotations better, hint hint
+set.seed(42)
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graoh 3 code}
+ggplot(mpg, aes(displ, hwy)) +
+  geom_point() +
+  geom_point(data = corvette, color = "blue") + #highlight only the corvette datapoints in blue
+  geom_text_repel(data = corvette, label = paste("Corvette,", corvette$year)) + #Label the highlighted datapoints using paste function.
+  labs(title = "Corvettes are a bit of an outlier") + #Add plot title
+  theme(plot.title = element_text(face = "bold")) + #make plot title bold
+  expand_limits(x=1:8)
+```
+
+There is a trick to getting the model and year to print off together. `paste()` is a useful function for this, also pasting together parts of file names and parts of urls together. 
+
+
+### Graph 4 
+```{r, echo = T}
+data(mpg)
+
+#hint for the coloring, colorbrewer and you can set palette colors and make your graphs colorblind friendly
+library(RColorBrewer)
+display.brewer.all(colorblindFriendly = T) #take a look at the colorblindfriendly options
+```
+
+The above graph lets you see some colobrlind friendly palettes. For the graph below, I used Set2.
+
+
+
+Now using the above mpg dataset, make this graph
+
+```{r graph 4 code}
+ggplot(mpg, aes(class, cty)) +
+  geom_boxplot(outlier.shape = NA) +
+  coord_flip() +
+  geom_jitter(size= 1.5, aes(color = class), height = 0) +
```

Comment:
This was probably the hardest graph for me to make. I couldn't figure out exactly how to use the geom_jitter function to make it look like the example. This looks exactly like the example, so good job.